tests/utils/audit_table_utils.py

```diff
@@ -67,3 +73,40 @@ def create_insert_query(
         audit=audit,
         parent_log_span=parent_log_span,
     )
+
+
+class QueryStringCriterion(Enum):
+    REGEXP = auto()
+    STARTS_WITH = auto()
+
+
+def expected_query(
+    table_name: TableName,
+    db_operation_type: DbOperationType = DbOperationType.INSERT,
+) -> Query:
+    query_strings = {
+        DbOperationType.INSERT: f"INSERT INTO {table_name.fully_qualified}",
+        DbOperationType.CREATE_IF_NOT_EXISTS: f"CREATE TABLE IF NOT EXISTS {table_name.fully_qualified}",
+    }
+    return ModifyQuery(
+        query_string=query_strings[db_operation_type],
+        db_object_type=DbObjectType.TABLE,
+        db_object_name=table_name,
+        db_operation_type=db_operation_type,
+    )
+
+
+def assert_queries_match(
```

Comment:
Did you register this as pytest custom assert? Alternative, I would suggest, either rewrite it as MatcherObject or register it as custom assert.

Reply:
https://lorepirri.com/pytest-register-assert-rewrite.html
https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=pytest%20register_assert_rewrite#pytest.register_assert_rewrite
https://docs.pytest.org/en/7.1.x/how-to/assert.html?highlight=pytest%20register_assert_rewrite#assertion-introspection-details